File: gaia-sdk-javascript/src/graphql/request/type/CreatedRoleImpulse.ts

import {Role} from "./Role";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * Impulse which indicates the result of a create role impulse
 */
export class CreatedRoleImpulse extends Array<(_:VariableRegistry) => string> {
public _typeName = "CreatedRoleImpulse";
    public id = () => { 
        this.push(_ => "id")
    };

    public data = (config: (_:Role) => void) => this.push((registry) => {
        const entity = new Role();
        config(entity);
        return "data { " + entity.render(registry) + " }";
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/request/type/CreatedRoleImpulse.py

from gaia_sdk.graphql.request.type.Role import Role

from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class CreatedRoleImpulse(list):
    """
    Impulse which indicates the result of a create role impulse
    """

    def id(self):
        self.append(lambda x: "id")

    def data(self, config: Callable[['Role'], None]):
        def callback(registry: VariableRegistry):
            entity = Role()
            config(entity)
            return "data {" + entity.render(registry) + "}"
        self.append(callback)

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))
